### DISCUSSION TITLE
HAOS uses way too many resources ?

### DISCUSSION THREAD

**Discussion by haldi4803** [home-assistant/operating-system#2972] [General]
Hello,

At first i tought i could run Home Assistant as a Docker Container on my Router. Quad Core A53 with 2gb RAM.
![image](https://github.com/home-assistant/operating-system/assets/17104473/8a0c247b-330f-48cf-af1b-4d5c774b3fda)
That worked pretty well until i realized i couldn't get any addons...

But hey, i also got my trustworthyNAS around, with X86 so no problem!
Intel Xeon E3-1265L with 16GB RAM, lil bit older but beefy enought i'd say right?
Running Xpenology (Synology on Custom Hardware) with VMM that should work with HAOS.
Downloaded the 11.2 ova file and imported that.

But as soon as i pressed "start" i got greeted by 56% CPU Load.
What? I set 7 from 8 threads because more is better... if it doesn't need them it should just idle right?
![image](https://github.com/home-assistant/operating-system/assets/17104473/b964b936-f8b8-42a0-9b83-ee947d8ba600)

I tought maybe that's just the early setup and the system will cool down a little bit later.... but no! It's been like that and staying like that forever.

To make sure it's not some kind of visual Error i've checked with htop and netdata on the Host side:
![image](https://github.com/home-assistant/operating-system/assets/17104473/ee4e16d9-07fb-4ec4-b311-9e0de2b112c9)

![image](https://github.com/home-assistant/operating-system/assets/17104473/48ca36ed-deb1-4227-b752-e1d54d0e63e7)

the CPU Usage inside HAOS is constantly around 30%
![image](https://github.com/home-assistant/operating-system/assets/17104473/ea3849a1-2f63-41b5-8cb9-fac60821255a)

the strange thing is that the 4GB limited memory from the Host side are fully used, but HAOS only reports 1.1GB RAM usage.

Running htop inside the Terminal of HAOS shows around 30% usage but no processes that are using as much.
![image](https://github.com/home-assistant/operating-system/assets/17104473/a4437a3c-3957-4ba9-8b53-47a363082123)

directly connecting to the VM i'm stuck in the ha CMD and cannot check anything, cannot quit that.
![image](https://github.com/home-assistant/operating-system/assets/17104473/62e2ee7d-72d8-477f-82ea-90acbaee67ac)

Is that Usage normal?
I've Rebooted in Safemode just to check, but saw no difference.
I do realize that HAOS is also running it's own docker system? and some other stuff so it does need more than the docker container.
But still, thats a little bit more usage than i would have expected.

How much does it use on your system?

**Comment by Impact123** [comment]
Type `login` into the `ha >` prompt to get to the shell. You can then investigate further with `top -o %CPU`.
You can also install the `glances` addon and press `Z` to show processes and then `c` to sort them by `CPU%`.
Id suggest you [check out the discord server](https://discord.gg/c5DvZ4e) for help with troubleshooting this further.

**Comment by haldi4803** [comment]
Yeah.... i don't exactly know what happend....
i've set it to 2 CPU cores for a while, now rebooted, set it 7 again and Everything works flawless as expected!
Around 1-2% CPU Usage!

![image](https://github.com/home-assistant/operating-system/assets/17104473/f51e5bde-7793-4aba-a9d8-85fa4ba37411)

![image](https://github.com/home-assistant/operating-system/assets/17104473/7d41b320-2f6c-4113-8731-f4c1fa10892a)

![image](https://github.com/home-assistant/operating-system/assets/17104473/aa47b8e6-6ff5-4820-bcc3-1694994a4b01)

![image](https://github.com/home-assistant/operating-system/assets/17104473/4a2c9ec5-68fb-4249-aa5e-b5742a2b8792)

I'm pretty sure the only thing i did at that point (12/12) was update 2 PLugins, one was SSH & Terminal and the other i can't remember right now.

But all is well if everything is well.
Seems like that truly was a bug / unwanted behaviour.

**Comment by haldi4803** [comment]
I think I found the issue.
hassio_dns is around 200% CPU load.
Happened again today after a few updates.
Restarted the whole Host System in which the VM runs. Afterwards CPU load was huge. Around 70% on the host.

Glances says that's all fault of the hassio_dns container.
![Screenshot_20240112-143530](https://github.com/home-assistant/operating-system/assets/17104473/84c49d0f-e492-4de7-8d11-ef10ed53e6a1)

**Reply by haldi4803** [reply]
https://community.home-assistant.io/t/high-cpu-load-hassio-dns-container/339584/19?u=haldi
